Sources/Currency/AnyCurrency.swift

```diff
@@ -14,70 +14,88 @@
 
 import Foundation
 
-/// A type-erased representation of a currency, with a single `Foundation.Decimal` holding the exact value.
+/// A type-erased representation of a currency.
 ///
 /// Conforming types will also provide `CurrencyMetadata` to access specific ISO 4217 information, such as the alphabetic code and "minor units".
 ///
-/// **Initialization**
+/// **Value Storage**
 ///
-/// As `Foundation.Decimal` has different precision to `BinaryFloatingPoint`, precision will be lost when initializing using float literals.
+/// All currencies have a "minorUnits" scale, as defined by their `CurrencyMetadata` that determine their precision when represented as a `Foundation.Decimal`.
 ///
-/// **Equality Comparisons**
+/// This "minorUnits" scale is the value that is stored within the currency for use in arithmetic and value comparisons.
 ///
-/// Floating point values are notorious for having precision variance, even when they are equivalent within the precision range a developer desires.
+/// **Initialization**
 ///
-/// To overcome this, all equality checks between two given currencies will use the `roundedAmount` by default.
+/// All values will be "bankers" rounded before being stored within the value. So a value of `USD(1.982)` will be stored as `1.98`.
```

Comment:
There were two main motivations that I should capture in the motivation blocks more concisely

1. Minor units is being used a lot for these calculations, and as is (without #9), is a computed property that does a lot of rounding and calculations, which is expensive.

It made more sense to just use that as the backing store, as it easily captures the precision of the currency and makes arithmetic faster than several floating point operations.

I noticed a 33% increase in performance of the two algorithms after switching `minorUnits` to be the storage from the `exactAmount: Decimal`.

2. `exactAmount` wasn't providing enough value to have around, with the chances of being a footgun. I originally had it as a way for developers to have the _exact_, exact precision of a currency - but I decided to hold the opinion that if a developer needs more precision than the ISO currencies specify - such as for US Gas being 3 digits rather than 2 - then they could easily write their own concrete type called `USGas` or something to gain that extra precision.

The complexity within the library isn't worth it. The `exactAmount` was too precise.

Reply:
As discussed, LGTM for now and we can always improve the accuracy later on.